addons/misra.py

```diff
@@ -566,8 +566,12 @@ def remove_file_prefix(file_path, prefix):
         result = file_path
     return result
 
-class Rule:
-    """ Class to keep rule text and metadata """
+
+class Rule(object):
+    """Class to keep rule text and metadata"""
+
+    SEVERITY_MAP = {'Required': 'warning', 'Mandatory': 'error', 'Advisory': 'style', 'style': 'style'}
```

Comment:
hmm.. this was not your idea, but this does not seem right to me. The severity should explain what type of issue is seen. Here, another definition is used.

I think it's better we always say 'style' for misra addon output. I did not want to put many (or all) of these checks in Cppcheck because they are too stylistic.